sdk/core/switcher/entry.S

```diff
@@ -302,6 +314,16 @@ after_zero:
 	// ca1, used for second return value
 	zeroAllRegistersExcept ra, sp, gp, s0, s1, a0, a1
 	cret
+
+	// If the stack is too small, we don't do the call, but to avoid leaking
+	// any other state we still go through the same return path as normal.  We
+	// set the return registers to -ENOTENOUGHSTACK and 0, so users can see
+	// that this is the failure reason.
+.Lstack_too_small:
+	// Return register is -ENOTENOUGHSTACK
+	li                 a0, -140
```

Comment:
Why can't we use the macro here?

Reply:
We probably could.  I wasn't sure about including errno.h, but I think it should be fine in hindsight.